docs/data-type-support.rst

```diff
@@ -0,0 +1,75 @@
+.. meta::
+   :description: hipcub API library data type support
+   :keywords: hipcub, ROCm, API library, API reference, data type, support
+
+.. _data-type-support:
+
+******************************************
+Data type support
+******************************************
+
+* Supported input and output types.
+
+  .. list-table:: Supported Input/Output Types
+    :header-rows: 1
+    :name: supported-input-output-types
+
+    *
+      - Input/Output Types
+      - Library Data Type
+      - AMD Support
+      - CUDA Support
+    *
+      - int8
+      - int8_t
+      - ✅
+      - ✅
+    *
+      - float8
+      - Not Supported
+      - ❌
+      - ❌
+    *
+      - bfloat8
+      - Not Supported
+      - ❌
+      - ❌
+    *
+      - int16
+      - int16_t
+      - ✅
+      - ✅
+    *
+      - float16
+      - __half
+      - ✅
+      - ✅ [#]_
+    *
+      - bfloat16      
+      - hip_bfloat16
+      - ✅
+      - ✅ [#]_
+    *
+      - int32
+      - int
+      - ✅
+      - ✅
+    *
+      - tensorfloat32
+      - Not Supported
+      - ❌
+      - ❌
+    *
+      - float32
+      - float
+      - ✅
+      - ✅
+    *
+      - float64
+      - double
+      - ✅
+      - ✅
+
+.. rubric:: Footnotes
+.. [#] NVidia backend can't handle ``half`` with the following API calls: ``block_adjacent_difference``, ``device_adjacenet_difference``, ``device_reduce``, ``device_scan``, ``device_segmented_reduce`` and ``device_select``.
+.. [#] NVidia backend can't handle ``bfloat16`` with the following API calls: ``block_adjacent_difference``, ``device_adjacenet_difference``, ``device_reduce``, ``device_scan``, ``device_segmented_reduce``, ``device_select`` and ``device_histogram``.
```

Comment:
These should probably be moved to "known issues", which is usually located in the changelog

Reply:
@MKKnorr I think we decided not to document known issues in the new changelog template,. Please check with @LisaDelaney

Reply:
Ok, but it shouldn't reside in the documentation, as this is something that can and most likely will change. On top of that, this is something we have no control over

Reply:
In my opinion we should not include such transient things in the documentation. Publishing versioned changelogs, are fine, as they don't change any more after release. Known issues however have to be constantly kept track of, when they are being included in the documentation, and require maintenance at some point in time, or else the documentation goes out of date. This is further explained in googles documentation style guide https://developers.google.com/style/timeless-documentation. I'm not sure if we follow that one, but it has some good intentions